Data_Engine/Query/Linq/FilterBy.cs

```diff
@@ -0,0 +1,63 @@
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+
+using BH.oM.Reflection.Attributes;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using BHRE = BH.Engine.Reflection;
+
+namespace BH.Engine.Data
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Filters a list of objects to output only those with a property value matching that input")]
+        [Input("objects", "List of objects to be filtered. All objects in the list should be of similar type")]
+        [Input("propertyName", "The name of the property to filter the objects by")]
+        [Input("value", "The value of the selected property to filter the objects by")]
+        public static List<T> FilterBy<T>(this List<T> objects, string propertyName, object value)
+        {
+            if (objects == null || objects.Count == 0) return null;
+            int groupedObjects = objects.GroupBy(x => x.GetType()).Count();
+            System.Type objectType = objects[0].GetType();
+            if (groupedObjects != 1)
+            {
+                BHRE.Compute.RecordError("All objects in the list to be sorted should be of similiar type.");
+                return null;
+            }
+            if (!propertyName.Contains("."))
+            {
+                System.Reflection.PropertyInfo prop = objects[0].GetType().GetProperty(propertyName);
+                if (prop != null)
+                    return objects.Where(x => prop.GetValue(x).Equals(value)).ToList();                
```

Comment:
`Equal` will not work if `prop.GetValue(x)` is null. Better to use `==`

Reply:
Had a problem with `==` is why I changed this to `.Equals.` I tested on a simple string and would not say the value (to search for) and the propertyValue were the same unless I used `.Equals`. I think related to the List<T> issue above? Not sure how to get around it...

Reply:
You might want to do your own comparison method that you call within the Where then. That way, you can return false, when null, use Contains for string, and == for the rest.